pkg/monitortests/authentication/nodefaultserviceaccountoperatorstests/monitortest.go

```diff
@@ -0,0 +1,191 @@
+package nodefaultserviceaccountoperatortests
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	configv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/monitortestframework"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type noDefaultServiceAccountChecker struct {
+	kubeClient kubernetes.Interface
+	cfgClient  *configv1.ConfigV1Client
+}
+
+func NewAnalyzer() monitortestframework.MonitorTest {
+	return &noDefaultServiceAccountChecker{}
+}
+
+// Cleanup implements monitortestframework.MonitorTest.
+func (n *noDefaultServiceAccountChecker) Cleanup(ctx context.Context) error {
+	return nil
+}
+
+var exceptions = []func(pod corev1.Pod) bool{
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "cluster-version-operator-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "downloads-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "etcd-guard-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "ingress-canary-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "kube-apiserver-guard-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "kube-controller-manager-guard-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "openshift-kube-scheduler-guard-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "monitoring-plugin-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "multus-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "networking-console-plugin-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "network-check-target-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "verify-all-openshiftcommunityoperators-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "verify-all-openshiftredhatmarketplace-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "verify-all-openshiftcertifiedoperators-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "verify-all-openshiftredhatoperators-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "version-")
+	},
+	func(pod corev1.Pod) bool {
+		return strings.HasPrefix(pod.Name, "must-gather-")
+	},
+	func(pod corev1.Pod) bool {
+		return pod.Namespace == "openshift-marketplace"
+	},
+}
+
+// generateDefaultSAFailures generates a list of failures where the pod in a list of pods
+// violated the default service account check.
+func generateDefaultSAFailures(podList []corev1.Pod) []string {
```

Comment:
Instead of returning a list of strings here, would it make sense for this to just return the full set of junit tests?